include/knnga.hpp

```diff
@@ -241,7 +241,7 @@ namespace Gamera { namespace GA {
 
             for (size_t i = 0; i < bestIndi.size(); ++i) {
                 this->knn->selection_vector[(*this->indexRelation)[i]] = bestIndi[i];
-                this->bestSolution[(*this->indexRelation)[i]] = bestIndi[i];
+                this->bestSolution[(*this->indexRelation)[i]] = (bool)bestIndi[i];
```

Comment:
Seems to be a reasonable guess, because the vector is initialized with false before.